### PR TITLE
feat: #4528 persist page serverside

### PR DESCRIPTION
**Pull request by joshbalfour**
# Pull request checklist

**Detail as per issue below (required):**

fixes: #4528

### CODE REVIEW

File: packages/app-builder-backend/package.json [hunk 1]
```diff
@@ -10,29 +10,34 @@
   "author": "Reapit Ltd",
   "main": "./src/index.ts",
   "scripts": {
-    "start": "ts-node .",
+    "start": "DYNAMODB_ENDPOINT=http://localhost:8000 ts-node .",
```
Comment (bashleigh) [inline packages/app-builder-backend/package.json:13]: config env?

File: packages/app-builder-backend/package.json [hunk 2]
```diff
@@ -10,29 +10,34 @@
...
     "test": "echo '...skipping...'",
     "build": "cp config.json src && tsc",
     "lint": "eslint --cache --ext=ts,tsx,js src --fix",
     "release": "DOCKER_BUILDKIT=1 cdk deploy --require-approval never",
     "publish": "yarn npm publish --access public",
     "conf": "yarn config-manager --namespace cloud --entity app-builder --mode fetch",
     "synth": "cdk synth",
+    "ddb": "docker run -d -p 8000:8000 amazon/dynamodb-local",
```
Comment (bashleigh) [inline packages/app-builder-backend/package.json:21]: docker-compose?

File: packages/app-builder-backend/src/ddb.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+import { Page } from './entities/page'
+
+const { APPS_TABLE_NAME = 'apps', GSI_NAME = 'userId-index', DYNAMODB_ENDPOINT, AWS_REGION = 'eu-west-2' } = process.env
+
+export const ddb = new DynamoDBClient({
+  endpoint: DYNAMODB_ENDPOINT,
+  region: AWS_REGION,
+})
+
+const getCreateTableCommand = (tableName: string): CreateTableCommand => {
```
Comment (bashleigh) [inline packages/app-builder-backend/src/ddb.ts:20]: Can this be done with CDK?

**Review by bashleigh — APPROVED** [review]
Approved with a few suggestions/little questions

File: packages/app-builder-backend/src/ddb.ts (hunk 3 above)
Comment (joshbalfour) [reply packages/app-builder-backend/src/ddb.ts:20]: this is only used locally, am using CDK in dev/prod

File: packages/app-builder-backend/package.json (hunk 1 above)
Comment (joshbalfour) [reply packages/app-builder-backend/package.json:13]: i would but because the lambda's using environment variables when deployed this felt cleaner

**Pull request merged**